### PR TITLE
Retrieve git repo url for a given build

### PR DESCRIPTION
**Pull request by mthak**
*Given a build get_repo_url shall return the git repo url of that build
* Implemented for git only as of now

### CODE REVIEW

File: jenkinsapi_tests/unittests/test_build_scm_git.py [hunk 1]
```diff
@@ -51,9 +51,9 @@ def test_git_revision_branch(build):
...
 
 
-def test_git_repo_url(self):
+def test_git_repo_url(build):
     """
     Can we extract git repo url for a given build
     """
-    self.assertIsInstance(self.b.get_repo_url(), six.string_types)
-    self.assertEquals(self.b.get_repo_url(), 'https://github.com/salimfadhley/jenkinsapi.git')
+    build.assertIsInstance(build.get_repo_url(), six.string_types)
```
Comment (lechat) [inline jenkinsapi_tests/unittests/test_build_scm_git.py]: We are not using `nosetests`, thus there is no such method as `assertIsInstance` or `assertEquals`. The test function above this new one can be used as example of `pytest` assertions.

Comment (mthak) [reply jenkinsapi_tests/unittests/test_build_scm_git.py]: ahh fixed and retriggered

File: jenkinsapi_tests/unittests/test_build_scm_git.py [hunk 2]
```diff
@@ -49,3 +49,13 @@ def test_git_revision_branch(build):
...
+
+
+def test_git_repo_url(build):
+    """
+    Can we extract git repo url for a given build
+    """
+    if not isinstance(build.get_repo_url(), str):
+        raise AssertionError
+    if not build.get_repo_url() == 'https://github.com/salimfadhley/jenkinsapi.git':
+        raise AssertionError
```
Comment (lechat) [inline jenkinsapi_tests/unittests/test_build_scm_git.py]: Why this is better than asserts in function above?

File: jenkinsapi/build.py [hunk 3]
```diff
@@ -180,6 +184,23 @@ def _get_git_rev_branch(self):
...
 
+    def _get_git_repo_url(self):
+        # Sometimes we have None as part of actions. Filter those actions
+        # which have lastBuiltRevision in them
+        _actions = [x for x in self._data['actions']
+                    if x and "lastBuiltRevision" in x]
+        # old Jenkins version have key remoteUrl v/s the new version has a list remoteUrls
+        result = _actions[0].get("remoteUrls", _actions[0].get("remoteUrl"))
+        if isinstance(result, list):
+            result = result[0]
```
Comment (lechat) [inline jenkinsapi/build.py]: what if I need to know all remote urls? Why not return list and let client decide?

Comment (mthak) [reply jenkinsapi/build.py]: for a given repo you will only have one git url. Unless we are using multiscm plugin in which case the api will differ. I am not sure why we changed remoteurl to remoteUrls in current api version .. I looked at the dict it returns and it's a list with one value all the time.

File: jenkinsapi_tests/unittests/test_build_scm_git.py (hunk 2 above)
Comment (mthak) [reply jenkinsapi_tests/unittests/test_build_scm_git.py]: I wrote the assert similar to one above , but the py coverage rejected it sayinh not meeting pylint standards.

File: jenkinsapi/build.py (hunk 3 above)
Comment (mthak) [reply jenkinsapi/build.py]: fixed
 if isinstance(result, list):
            result = ','.join(result)

File: jenkinsapi_tests/unittests/test_build_scm_git.py (hunk 2 above)
Comment (lechat) [reply jenkinsapi_tests/unittests/test_build_scm_git.py]: Please revert it to normal pytest assertions. The warning that asserts will be stripped out doesn't apply in this case.

Comment (mthak) [reply jenkinsapi_tests/unittests/test_build_scm_git.py]: done

Comment (mthak) [reply jenkinsapi_tests/unittests/test_build_scm_git.py]: The build failed with checks again when i reverted the changes
Showing 1 file with new issues 
jenkinsapi_tests/unittests/test_build_scm_git.py . Below is the error

Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
58
assert isinstance(build.get_repo_url(), str)
Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
59
assert build.get_repo_url() == 'https://github.com/salimfadhley/jenkinsapi.git'
Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.
60
assert len(build.get_repo_url()) == 1
image

File: jenkinsapi/build.py (hunk 3 above)
Comment (mthak) [reply jenkinsapi/build.py]: I changed the tests to assert Condition ... and the code coverage is back to red with that change .. but can you please look at it and merge it.

**Pull request merged**